apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.CommonTesting/SpannerTestDatabase.cs

```diff
@@ -110,6 +117,40 @@ private SpannerTestDatabase(string projectId)
             }
         }
 
+        private void CreateInstanceOnEmulatorIfNeeded(string projectId)
+        {
+            // Try to create an instance on the emulator and ignore any AlreadyExists error.
+            var adminClientBuilder = new InstanceAdminClientBuilder
+            {
+                EmulatorDetection = EmulatorDetection.EmulatorOnly
+            };
+            var instanceAdminClient = adminClientBuilder.Build();
+
+            InstanceName instanceName = InstanceName.FromProjectInstance(projectId, SpannerInstance);
+            try
+            {
+                var operation = instanceAdminClient.CreateInstance(new CreateInstanceRequest
+                {
+                    InstanceId = instanceName.InstanceId,
+                    Parent = $"projects/{instanceName.ProjectId}",
+                    Instance = new Instance
+                    {
+                        InstanceName = instanceName,
+                        ConfigAsInstanceConfigName = new InstanceConfigName(projectId, "emulator-config"),
+                        DisplayName = "Test Instance",
+                        NodeCount = 1,
+                    },
+                });
+            }
```

Comment:
You need to call `operation.PollUntilCompleted()` so we wait for the instance to actually be created. (It may happen immediately on the emulator, but we shouldn't assume that.)

Reply:
Done